project_leverage/controllers/main.py

```diff
@@ -0,0 +1,109 @@
+# Copyright (C) 2020 Open Source Integrators
+# Copyright (C) 2020 Serpent Consulting Services
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo.http import request
+from odoo.exceptions import UserError
+
+from odoo.addons.sale_timesheet.controllers.main import SaleTimesheetController
+
+
+DEFAULT_MONTH_RANGE = 3
+
+
+class SaleTimesheetController(SaleTimesheetController):
+
+    def _plan_prepare_values(self, projects):
+
+        employee_obj = request.env['hr.employee']
+        values = super(SaleTimesheetController,
+                       self)._plan_prepare_values(projects)
+        _select = self.select_line()
+        _from = self.line_from()
+        _where = self.line_where(projects.ids)
+        _group = self.line_group()
+        query = _select + _from + _where + _group
+        request.env.cr.execute(query)
+        raw_data = request.env.cr.dictfetchall()
+        project_data = {}
+        month_list = []
+        month_name_list = []
+        for rec in raw_data:
+            if str(rec.get('month')).strip() + ' ' + str(
+                    int(rec.get('year'))) not in month_name_list:
```

Comment:
If you add a "month_year" SQL column, you won't need to repeat these string operation all these times.